src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,165 @@
+import { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { updateTodo, deleteTodos } from '../api/todos';
+import { showError } from '../helpers/helpers';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  setError: React.Dispatch<React.SetStateAction<string>>,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  loader: number[],
+  setLoader:React.Dispatch<React.SetStateAction<number[]>>
+  todo: Todo,
+}
+
+export const TodoInfo:React.FC<Props> = ({
+  setError,
+  setTodos,
+  loader,
+  setLoader,
+  todo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [searchQuery, setSearchQuery] = useState(todo.title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [isEditing]);
+
+  const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+  };
+
+  const clickHandler = (id: number) => {
+    setLoader(prevState => [...prevState, id]);
+    deleteTodos(id)
+      .then(() => {
+        setTodos(prevState => (
+          prevState.filter(prevTodo => prevTodo.id !== id)
+        ));
+      })
+      .catch(() => showError('Unable to delete a todo', setError));
+  };
+
+  const statusClickHandler = (todoId : number, todoStatus: boolean) => {
+    setLoader(prevState => [...prevState, todoId]);
+    updateTodo(todoId, { completed: !todoStatus })
+      .then((updatedTodo) => {
+        setTodos(prevState => (
+          prevState.map(prevTodo => {
+            return (prevTodo.id === todoId) ? updatedTodo : prevTodo;
+          })
+        ));
+
+        setLoader(prevState => prevState.filter(id => id !== todoId));
+      })
+      .catch(() => showError('Unable to delete a todo', setError));
+  };
+
+  const blurHandler = () => {
+    setIsEditing(false);
+
+    switch (searchQuery) {
+      case '':
+        setLoader(prevState => [...prevState, todo.id]);
+        deleteTodos(todo.id)
+          .then(() => {
+            setTodos(prevState => (
+              prevState.filter(prevTodo => prevTodo.id !== todo.id)
+            ));
+          })
+          .catch(() => showError('Unable to delete a todo', setError));
```

Comment:
how about we put this inside of a function? because we use this code twice in this component

DRY

Reply:
We don't use it twice, there are slight differences
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/130560419/1d6c09bd-ded1-4fc8-9816-428cf5f3158b)

Reply:
we can reuse only deleting handler beacuse in updating handler in first case we change status, in second case we change title